tools/diamond.py

```diff
@@ -99,7 +99,10 @@ def execute(self, command, options=None, option_string=None, return_stdout=False
         if option_string:
             cmd.extend(shlex.split(option_string))
         log.debug("Calling {}: {}".format(command, " ".join(cmd)))
-        return util.misc.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        res = util.misc.run_and_print(cmd)
```

Comment:
two thoughts:

1. why run_and_print and not subprocess.check_call?
2. if run_and_print, how about log.error(res.stdout.decode('utf-8')) instead of print()?